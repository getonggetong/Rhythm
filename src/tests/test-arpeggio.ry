/*	Arpeggio Test

	Tests the following:
	1. Algorithms in Rhythm
	2. Loop loop to build track
	3. If the songs are building with the correct notes
*/

init() {
	c = C4;
	song = [];
	loop (0 : 4) {
		song ::= c;
		c += 4;
	}
	song[2]--;
	song ::= song[2] :: song[1] :: song[0];
	print(song);
}