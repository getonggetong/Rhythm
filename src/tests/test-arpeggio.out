[C4,E4.4,G4.4,C5.4,G4.4,E4.4,C4]